### PR TITLE
Blog/adding assets

### PR DESCRIPTION
**Pull request by iobrado**
# Description

Another blog post which covers using assets like images or icons in your project. As always, suggestions are always welcome, this time for title and headings especially 😅 

If there is anything else you think should be covered here, let me know.

### CODE REVIEW

File: website/blog/2022-04-25-using-assets.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+title: Using assets in a project
+description: Step-by-step guide on how to add assets like images or icons to your theme.
+slug: using-assets
+authors: iobrado
+date: 2022-05-02
+tags: [eightshift, boilerplate, assets, images, icons]
+hide_table_of_contents: false
+---
+
+Previously we went through the process of adding fonts to your project. While the process of adding additional assets like images or icons has some similarities to adding fonts, it also has its unique steps. In this post, we'll cover multiple ways of adding assets and using them on your site.
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:11]:
```suggestion
Previously, we went through the process of adding fonts to your project. While the process of adding additional assets like images or icons has some similarities to adding fonts, it also has its unique steps. In this post, we'll cover multiple ways of adding assets and using them on your site.
```

File: website/blog/2022-04-25-using-assets.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+apply_filters(Manifest::MANIFEST_ITEM, 'logo.svg');
+```
+
+The filter we are using is called `manifest-item` and we use it to get the URL of the asset from the **_public_** folder. You can read more about this in [our documentation](/docs/basics/manifest).
+
+You can see how this is being used for rendering both favicon and header logo in your theme's **_header.php_** file.
+
+> **Tip**: Don't hardcode the filter name in the `apply_filters` function. Always call it via class constants.
+
+For better organization, you can add additional folders (e.g. **_icons_**, **_placeholders_**) inside **_assets/images_** folder. Here's an example how to include them:
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:36]:
```suggestion
For better organization, you can add additional folders (e.g. **_icons_**, **_placeholders_**) inside the **_assets/images_** folder. Here's an example of how to include them:
```

File: website/blog/2022-04-25-using-assets.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import './placeholders/page.png';
+```
+
+## Using SVG files from manifest
+
+Another approach you can take is to include an SVG file inside your component or block manifest. As a matter of fact, we already mentioned that component in a previous blog post about [Modifying Blocks](/blog/modifying-blocks-color-theme). We are talking about Quote component.
+
+> If you don't have it in your project, be sure to read our blog post about adding blocks and components by using [WP CLI](/blog/adding-blocks-wpcli).
+
+Open **_src/Blocks/components/quote/manifest.json_** and you'll see that all icons used by the component are defined inside `icons` as key-value pairs. Key represents the name which we will use to fetch the icon, while the value is SVG code.
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:53]:
```suggestion
Open **_src/Blocks/components/quote/manifest.json_** and you'll see that all icons used by the component are defined inside `icons` as key-value pairs. Key represents the name that we will use to fetch the icon, while the value is SVG code.
```

File: website/blog/2022-04-25-using-assets.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+```json
+"resources": {
+	"icon": "<svg fill='none' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'>...</svg>"
+}
+```
+
+In order to make minification of SVG files as easy as possible, our teammate Goran made an awesome tool called [SVG2WP](https://svg-2-wp.goranalkovic.com/). Some of the options include making attributes JSX compatible, or replacing the color value with `currentColor`, which can then be used to change the SVG color through CSS.
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:61]:
```suggestion
In order to make the minification of SVG files as easy as possible, our teammate Goran made an awesome tool called [SVG2WP](https://svg-2-wp.goranalkovic.com/). Some of the options include making attributes JSX compatible, or replacing the color value with `currentColor`, which can then be used to change the SVG color through CSS.
```

File: website/blog/2022-04-25-using-assets.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+If you've been reading through these blog posts regularly, then you've already seen the use of `currentColor`. In our blog post about [Modifying Blocks](/blog/modifying-blocks-color-theme), we modified the color of the SVG.
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:63]:
```suggestion
You've already seen the use of `currentColor` in the above-mentioned blog post, where we've modified the color of the SVG.
```

File: website/blog/2022-04-25-using-assets.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+The output of the icon on frontend is very simple. In the Quote component, it was done the following way:
+```php
+<?php $manifest = Components::getManifest(__DIR__); ?>
+
+<i class="<?php echo \esc_attr("{$componentClass}__icon"); ?>">
+	<?php echo $manifest['resources']['icon']; // phpcs:ignore Eightshift.Security.ComponentsEscape.OutputNotEscaped ?>
+</i>
+```
+
+An excellent example where you can see in even more detail how SVGs are being used is our `icon` component. It isn't included in Eightshift theme by default, so you have to add it to your project with WP CLI. To include it in your project, use the following command:
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:74]:
```suggestion
An excellent example, where you can see in even more detail how SVGs are being used, is our `icon` component. It isn't included in Eightshift theme by default, so you have to add it to your project with WP CLI. To include it in your project, use the following command:
```

File: website/blog/2022-04-25-using-assets.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+```php
+<?php
+echo Components::render(
+	'icon',
+	[
+		'blockClass' => $componentClass,
+		'iconName' => 'download',
+	]
+);
+?>
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:91]:
```suggestion
```

Always omit the closing PHP tag, unless inside HTML block

File: website/blog/2022-04-25-using-assets.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+Here are some examples of icons available out-of-the-box in our Icon component:
+
+![Icon component](/img/blog/icon-component.png)
+
+## Using icons for editor and block options
+
+When developing your blocks and adding new options, you may need to add some icons to improve the user experience. We have many icons already available for use. You can see the full list in our [Storybook](/storybook) under `Editor -> Icons` section. We already added the icon when adding a new Color Theme option for the Quote block. Here is the simplified version:
```
Comment (dingo-d) [inline website/blog/2022-04-25-using-assets.md:100]:
```suggestion
When developing your blocks and adding new options, you may need to add icons to improve the user experience. We have many icons already available for use. You can see the full list in our [Storybook](/storybook) under `Editor -> Icons` section. We already added the icon when adding a new Color Theme option for the Quote block. Here is the simplified version:
```

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left some minor comments, great work!

File: website/blog/2022-04-25-using-assets.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+import './icons/upload.svg';
+
+// Placeholders
+import './placeholders/post.png';
+import './placeholders/page.png';
+```
+
+## Using SVG files from manifest
+
+Another approach you can take is to include an SVG file inside your component or block manifest. As a matter of fact, we already mentioned that component in a previous blog post about [Modifying Blocks](/blog/modifying-blocks-color-theme). We are talking about Quote component.
```
[the remark this answers is not in the document]
Comment (iobrado) [reply website/blog/2022-04-25-using-assets.md:49]: Oh, this is a great suggestion! 💯

**Review by dingo-d — APPROVED** [review]

File: website/blog/2022-04-25-using-assets.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+import './placeholders/page.png';
+```
+
+## Using SVG files from manifest
+
+If you recall from a previous blog post about [Modifying Blocks](/blog/modifying-blocks-color-theme), you might have already seen an alternative approach to including SVG files in your block or component.
+
+> If you don't have it in your project, be sure to read our blog post about adding blocks and components by using [WP CLI](/blog/adding-blocks-wpcli).
+
+Open **_src/Blocks/components/quote/manifest.json_** and you'll see that all icons used by the component are defined inside `icons` as key-value pairs. Key represents the name that we will use to fetch the icon, while the value is SVG code.
```
[the remark this answers is not in the document]
Comment (iobrado) [reply website/blog/2022-04-25-using-assets.md:53]: Ha, great catch! 😄 
Initially I started explaining this with `icon` component, but then I decided to rearrange the content a bit. This is a leftover from that. I'll update it accordingly.

**Review by mbmjertan — APPROVED** [review]

**Pull request merged**